buildtools/AWS.Deploy.Tools.CI/AWS.Deploy.Tools.CI/CodeBuildFunctions.cs

```diff
@@ -0,0 +1,123 @@
+using Amazon.SecurityToken;
+using Amazon.SecurityToken.Model;
+using Amazon.CodeBuild;
+using Amazon.CodeBuild.Model;
+using Amazon.Lambda.Core;
+using Amazon.Lambda.Annotations;
+using Amazon.Lambda.Annotations.APIGateway;
+using AWS.Deploy.Tools.CI.Models;
+
+[assembly: LambdaSerializer(typeof(Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer))]
+
+namespace AWS.Deploy.Tools.CI;
+
+/// <summary>
+/// Lambda functions that will interact with CodeBuild as part of the AWS.Deploy.Tools CI/CD pipeline.
+/// </summary>
+public class CodeBuildFunctions
+{
+    /// <summary>
+    /// The <see cref="GetAvailableTestRunner"/> function is responsible for checking available test runner accounts to run the AWS.Deploy.Tools CodeBuild PR check.
+    /// A list of IAM roles, representing the test runner accounts, is passed to the function. 
+    /// These roles are assumed and used to check if the CodeBuild CI project in the test runner account is currently running any jobs.
+    /// </summary>
+    /// <returns>The function will return the IAM role of the account that is not running any CodeBuild CI jobs.</returns>
+    /// <exception cref="ArgumentNullException">If the input passed to the function is invalid.</exception>
+    /// <exception cref="Exception">If no test runner account is available.</exception>
+    /// <exception cref="Exception">If a CodeBuild CI project is not found in the test runner account.</exception>
+    [LambdaFunction(Name = "GetAvailableTestRunner", Policies = "@AWSDeployToolsCIGetAvailableTestRunnerLambdaAssumeRolePolicy")]
+    public async Task<string> GetAvailableTestRunner([FromBody] GetAvailableTestRunnerInput input)
```

Comment:
The `[FromBody]` doesn't do anything here because this is not an API Gateway based function.

Reply:
Done